vscode/src/debugger/session.ts

```diff
@@ -390,25 +390,34 @@ export class QscDebugSession extends LoggingDebugSession {
     // This will be replaced when the interpreter
     // supports shots.
     for (let i = 0; i < args.shots; i++) {
-      const result = await this.debugService.evalContinue(
-        bps,
-        this.eventTarget,
-      );
-      if (result.id != StepResultId.Return) {
-        await this.endSession(`execution didn't run to completion`, -1);
-        return;
-      }
-      this.writeToDebugConsole(`Finished shot ${i + 1} of ${args.shots}`);
-      // Reset the interpreter for the next shot.
-      // The interactive interpreter doesn't do this automatically,
-      // and doesn't know how to deal with shots like the stateless version.
-      await this.init(associationId);
-      if (this.failureMessage != "") {
-        log.info(
-          "compilation failed. sending error response and stopping execution.",
+      const success = await this.debugService
+        .evalContinue(bps, this.eventTarget)
+        .then(
```

Comment:
Mixing `await` and `then` patterns is pretty confusing - could you rewrite this as just straight line code with `await`s? Errors can simply be `try`/`catch`ed.